.github/workflows/ci.yml

```diff
@@ -0,0 +1,80 @@
+name: CI
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+env:
+  app: sidekiq_publisher
+
+jobs:
+  lint:
+    name: Lint (Ruby ${{ matrix.ruby }})
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [ '3.0', '3.1', '3.2' ]
+
+    steps:
+      - name: Checkout the code
+        uses: actions/checkout@v3
+
+      - name: Install Ruby ${{ matrix.ruby }}
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+
+      - name: Install dependencies
+        run: bundle install
+
+      - name: Run Rubocop
+        run: bundle exec rubocop
+
+  test:
+    name: Test (Ruby ${{ matrix.ruby }}, Redis ${{matrix.redis}}, Postgres ${{ matrix.postgres }})
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [ '3.0', '3.1', '3.2' ]
+        postgres: [ '12', '13', '14', '15', '16' ]
+        redis: [ '6', '7' ]
```

Comment:
Curious if you think we need to test all the PG and Redis versions. I don't think we were before. Are we using anything beyond normal PG queries or Redis commands? Either way I think we can probably drop PG 12 and 13 at least, since the monolith along with other projects being on 14 now.

Reply:
Tbh this is almost certainly overkill; I was on a bit of a "Test All The Things" spree yesterday, given how many weird version issues we ran into with moving this project to GHA in the first place, but you make a good point re: redis and postgres queries

Reply:
I think I will tell the CI workflow to use `latest` of each and that way we'll see on any new PR if something breaks when one of the images gets updates